packages/react-test-renderer/src/ReactTestRendererAct.js

```diff
@@ -141,7 +144,8 @@ function act(callback: () => Thenable) {
         if (__DEV__) {
           warningWithoutStack(
             false,
-            'Do not await the result of calling act(...) with sync logic, it is not a Promise.',
+            'Do not await the result of calling a synchronous act(...), it is not a Promise. \n' +
```

Comment:
Do we even need this warning anymore?

Reply:
Also I would have done this in a separate PR to avoid giving me the opportunity to nitpick about it in this one :D

Reply:
[Using await for a non-promise should be fine](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#Description). If there is some other issue with this it should be included in the warning message. In its current form it is misleading.

Reply:
What would the behaviour be if `await`ed? Should it run flushPassiveEffectsAndMicrotasks? I... guess it should?

Reply:
In any case, this shouldn't block this PR, I'll do it in a followup PR if needed, and if critical for 16.9, fair?

Reply:
This is a little harder than I expected. sync act() cleans up on exiting, so if any updates are fired on awaiting a sync act, they'll trigger the missing act() warning. using the async () => version seems cleanest.

Reply:
I started work on this in a separate branch, but it's tricky so far. awaiting inserts one microtask and could miss an update. failing code here rn https://github.com/threepointone/react/tree/await-sync-act will keep poking at it separately.